src/check/property/AsyncProperty.generic.ts

```diff
@@ -44,16 +47,28 @@ export interface IAsyncPropertyWithHooks<Ts> extends IAsyncProperty<Ts> {
  * @internal
  */
 export class AsyncProperty<Ts> implements IAsyncPropertyWithHooks<Ts> {
-  static dummyHook: AsyncPropertyHookFunction = () => {};
-  private beforeEachHook: AsyncPropertyHookFunction = AsyncProperty.dummyHook;
-  private afterEachHook: AsyncPropertyHookFunction = AsyncProperty.dummyHook;
-  constructor(readonly arb: Arbitrary<Ts>, readonly predicate: (t: Ts) => Promise<boolean | void>) {}
+  static dummyGlobalHook: GlobalAsyncPropertyHookFunction = () => {};
+  static defaultHook: AsyncPropertyHookFunction = (globalHook) => globalHook();
+  private readonly globalBeforeEachHook: GlobalAsyncPropertyHookFunction;
+  private readonly globalAfterEachHook: GlobalAsyncPropertyHookFunction;
+  private beforeEachHook: AsyncPropertyHookFunction = AsyncProperty.defaultHook;
+  private afterEachHook: AsyncPropertyHookFunction = AsyncProperty.defaultHook;
+  constructor(readonly arb: Arbitrary<Ts>, readonly predicate: (t: Ts) => Promise<boolean | void>) {
+    const {
+      asyncBeforeEach,
+      asyncAfterEach,
+      beforeEach = AsyncProperty.dummyGlobalHook,
+      afterEach = AsyncProperty.dummyGlobalHook,
+    } = readConfigureGlobal() || {};
+    this.globalBeforeEachHook = asyncBeforeEach || beforeEach;
+    this.globalAfterEachHook = asyncAfterEach || afterEach;
+  }
```

Comment:
As users are now powerful enough to tell the framework how to merge multiple hooks together, I would make no differences between global and local hooks. And then let the user does what she/he wants with consecutive declarations of beforeEach. In other words, the global hook is just used to initialize beforeEachHook but it does not have any specific priviledges.

I believe a code like that:
```js
configureGlobal({ asyncBeforeEach: () => console.log('1') })

fc.asyncProperty(...)
  .beforeEach(prev => {
    prev();
    console.log('2');
  })
  .beforeEach(prev => {
    prev();
    console.log('3');
  })
  .beforeEach(prev => {
    console.log('4');
    prev();
    console.log('5');
  })
```

Should resulting into a hook that would do:
```js
() => {
    console.log('4');
    console.log('1');
    console.log('2');
    console.log('3');
    console.log('5');
}
```

In other words, I'd update the initialization part as follow:

```suggestion
  static dummyHook: AsyncPropertyHookFunction = (globalHook) => globalHook();
  private beforeEachHook: AsyncPropertyHookFunction;
  private afterEachHook: AsyncPropertyHookFunction;
  constructor(readonly arb: Arbitrary<Ts>, readonly predicate: (t: Ts) => Promise<boolean | void>) {
    const { asyncBeforeEach,  asyncAfterEach,  beforeEach,  afterEach  } = readConfigureGlobal() || {};
    if (asyncBeforeEach !== undefined && beforeEach !== undefined) {
      // Concerning those checks they need to be introduce as soon as we release the version.
      // As a library authors, it's easier to relax constraints that add new ones (breaking change)
      throw new Error('Illegal');
    }
    if (asyncAfterEach !== undefined && afterEach !== undefined) {
      throw new Error('Illegal');
    }
    this.beforeEachHook = asyncBeforeEach || beforeEach || AsyncProperty.defaultHook;
    this.afterEachHook = asyncAfterEach || afterEach || AsyncProperty.defaultHook;
  }
```

The beforeEach method would then have to be updated as follow:
```js
  beforeEach(hookFunction: AsyncPropertyHookFunction): AsyncProperty<Ts> {
    const previousBeforeEachHook = this.beforeEachHook;
    this.beforeEachHook = () => hookFunction(previousBeforeEachHook);
    return this;
  }
```

Reply:
Oh, okay. I didn't know you wanted to go this route. Sounds good to me! :slightly_smiling_face: